crates/wdl-engine/src/inputs.rs

```diff
@@ -888,46 +885,60 @@ impl Inputs {
     ///
     /// Returns `Ok(None)` if the inputs are empty.
     pub fn parse_object(document: &Document, object: JsonMap) -> Result<Option<(String, Self)>> {
-        // Determine the root workflow or task name
-        let (key, name) = match object.iter().next() {
-            Some((key, _)) => match key.split_once('.') {
-                Some((name, _remainder)) => (key, name),
-                None => {
-                    bail!(
-                        "invalid input key `{key}`: expected the value to be prefixed with the \
-                         workflow or task name",
-                    )
-                }
-            },
-            // If the object is empty, treat it as a workflow evaluation without any inputs
-            None => {
-                return Ok(None);
-            }
+        // If the object is empty, treat it as an invocation without any inputs.
+        if object.is_empty() {
+            return Ok(None);
+        }
+
+        // Otherwise, build a set of candidate entrypoints from the prefixes of each
+        // input key.
+        let mut entrypoint_candidates = BTreeSet::new();
+        for key in object.keys() {
+            let Some((prefix, _)) = key.split_once('.') else {
+                bail!(
+                    "invalid input key `{key}`: expected the value to be prefixed with the \
```

Comment:
```suggestion
                    "invalid input key `{key}`: expected the key to be prefixed with the \
```

The term _value_ is confusing to me in the context of a key-value map (which is what an inputs JSON file is). Given that we used _key_ earlier in the message, I think using _key_ again is unambiguous whereas _value_ might mean either the value of the key or the value mapped by the key.